aws/resource_aws_codepipeline.go

```diff
@@ -605,8 +610,33 @@ func resourceAwsCodePipelineDelete(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting Codepipeline (%s): %s", d.Id(), err)
+		return fmt.Errorf("error deleting CodePipeline (%s): %w", d.Id(), err)
 	}
 
 	return err
 }
+
+func suppressCodePipelineStageActionConfiguration(k, old, new string, d *schema.ResourceData) bool {
+	parts := strings.Split(k, ".")
+	parts = parts[:len(parts)-2]
+	providerAddr := strings.Join(append(parts, "provider"), ".")
+	provider := d.Get(providerAddr).(string)
+
+	if provider == CodePipelineProviderGitHub && strings.HasSuffix(k, CodePipelineGitHubActionConfigurationOAuthToken) {
+		hash := hashCodePipelineGitHubToken(new)
+		return old == hash
+	}
+
+	return false
+}
+
+const codePipelineGitHubTokenHashPrefix = "hash-"
+
+func hashCodePipelineGitHubToken(token string) string {
```

Comment:
We are actively in the process of [removing the few other instances of hashed state storage](https://github.com/terraform-providers/terraform-provider-aws/issues/13406), however Terraform in general doesn't handle `TypeMap` and `Sensitive: true` in a meaningful way without making the entire map sensitive, if I remember correctly. 😖 

It would be a bummer to introduce this technical debt in, but otherwise we could also be forced down the road of what I think you were suggesting the other day of potentially going against the API design slightly and implementing bespoke Terraform schema so we can control it better (and in this situation, actually implement a single sensitive attribute). We do have some slight precedence of this class of user experience enhancement with resources like `aws_sns_platform_application` and I believe `aws_sns_topic`.

---

I think this leaves us in a tough spot. 🙁   We can try to quickly design something better for this situation to squeeze it in for this major release in the next two weeks, which is fairly risky in my opinion, or punt on this for now and spend later cycles on this issue in general (potentially changing the Core/SDK sensitive map abilities and/or ensuring we get good feedback on a new resource design). What do you think?

Reply:
I think the environment variable is a huge pain-point for practitioners, and waiting until 4.0 to remove it would be less than ideal. If we don't want to hash the value, would setting the `TypeMap` to `Sensitive` be a better option at this point? And then we could come up with a better solution.

Reply:
Our potential path forward will be:

* Removing the state hashing handling
* Marking the `configuration` attribute `Sensitive: true` and `Computed: true`
* Adding a new `github`/`github_configuration` configuration block so we can properly show a difference with only the `oauth` attribute sensitive

These will require apply-time validation to ensure they aren't both defined since `ConflictsWith` does not support relative references. The other configuration blocks will need to be filled in later due to time. The documentation and upgrade guide should recommend using the new configuration block(s) over the map.